helmrelease/release.go

```diff
@@ -0,0 +1,467 @@
+		} else if err != nil {
+			return microerror.Mask(err)
+		}
+
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("created release %#q", name))
+	}
+
+	if isOperator {
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("ensuring operator pod exists for release %#q", name))
+
+		c := r.Condition().PodExists(ctx, r.namespace, fmt.Sprintf("app=%s", name))
+		err := r.waitForConditions(ctx, c)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("ensured operator pod exists for release %#q", name))
+	}
+
+	var operatorPodName string
+	var operatorPodNamespace string = r.namespace
+	if isOperator {
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("finding operator pod name for release %#q", name))
+
+		operatorPodName, err = r.podName(operatorPodNamespace, fmt.Sprintf("app=%s", name))
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("found operator pod name for release %#q", name))
+	}
+
+	if isOperator {
+		err := r.fileLogger.EnsurePodLogging(ctx, operatorPodNamespace, operatorPodName)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+	}
+
+	{
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("ensuring conditions for release %#q", name))
+
+		err := r.waitForConditions(ctx, conditions...)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("ensured conditions for release %#q", name))
+	}
+
+	return nil
+}
+
+func (r *Release) Install(ctx context.Context, name string, chartInfo ChartInfo, values string, conditions ...ConditionFunc) error {
+	releaseName := fmt.Sprintf("%s-%s", r.namespace, name)
+
+	var err error
+
+	tarballPath, err := r.pullTarball(ctx, name, chartInfo)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	err = r.helmClient.InstallReleaseFromTarball(ctx, tarballPath, r.namespace, helm.ReleaseName(releaseName), helm.ValueOverrides([]byte(values)), helm.InstallWait(true))
+	if helmclient.IsReleaseAlreadyExists(err) {
+		return microerror.Maskf(releaseAlreadyExistsError, "failed to install release %#q", releaseName)
+	} else if helmclient.IsTarballNotFound(err) {
+		return microerror.Maskf(tarballNotFoundError, "failed to install release %#q from tarball %#q", releaseName, tarballPath)
+	} else if err != nil {
+		return microerror.Mask(err)
+	}
+
+	err = r.waitForConditions(ctx, conditions...)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	return nil
+}
+
+// TODO Remove once we are done with migration to EnsureInstalled.
+//
+//	Issue https://github.com/giantswarm/giantswarm/issues/4355.
+//
+func (r *Release) InstallOperator(ctx context.Context, name string, chartInfo ChartInfo, values string, crd *apiextensionsv1beta1.CustomResourceDefinition) error {
```

Comment:
I'd like to drop this. EnsureInstalled will handle that and we shouldn't install operators with `InstallOperator`.

Reply:
Removed.